### ISSUE TITLE
BizHawk no longer distinguishes between Left and Right Shift/Control keys

### ISSUE THREAD

**Issue by TheMetroid7998** [TASEmulators/BizHawk#3184]
edit: blocked by #2981

Summary: When setting controls for any console/game system, BizHawk no longer distinguishes between Left Shift & Right Shift, as well as Left Control and Right Control. They are instead simply merged into Shift and Ctrl, as compared to RShift/LShift & RCtrl/LCtrl.

Reproduction:
1. Literally just try to bind any controls to a keyboard on the current version.

BizHawk 2.8 Windows 10 Home Edition (x64), Intel CPU&GPU/NVIDIA Dedicated GPU

**Comment by YoshiRulz** [comment]
I guess you mean tapping Shift, as modifier keys were already merged. Unfortunately, while this is configurable with `MergeLAndRModifierKeys`, I put in this line:
https://github.com/TASEmulators/BizHawk/blob/107c7b703427c70ba112553b3f9166ba48a4d554/src/BizHawk.Client.EmuHawk/Input/Input.cs#L45
But I think there is a workaround. It only runs when EmuHawk first loads. You should be able to use `Config` > `Load Config` after manually editing your config to set `MergeLAndRModifierKeys: false`.

**Comment by TheMetroid7998** [comment]
Cool, that works great actually as most of my SNES games use totally different control schemes. Thanks for your help!

**Comment by Meerkov** [comment]
I was using Right Shift as an alternative for Pause on my laptop (which has no Pause key / has a smaller keyboard). Any chance this behavior can be the default when you're not using a button combination?

**Comment by TheMetroid7998** [comment]
Not with my keyboard...

**Comment by TheMetroid7998** [comment]
> I guess you mean tapping Shift, as modifier keys were already merged. Unfortunately, while this is configurable with `MergeLAndRModifierKeys` (not exposed in the UI), I put in this line:
> 
> https://github.com/TASEmulators/BizHawk/blob/107c7b703427c70ba112553b3f9166ba48a4d554/src/BizHawk.Client.EmuHawk/Input/Input.cs#L45
> 
> 
> But I think there is a workaround. It only runs when EmuHawk first loads. You should be able to use `Config` > `Load Config` after manually editing your config to set `MergeLAndRModifierKeys: false`.

So I found a new issue. Using the shift keys only works for one input at a time. 
e.g. holding L in Metroid Zero Mission to aim diagonally only aims Samus's cannon diagonally up for one beam shot, all other consecutive presses of B leave the cannon in a neutral position before returning to diagonal aim. Yeah, it's just as weird as it sounds.

**Comment by Meerkov** [comment]
To be clear, I'm wondering if this can return to the previous behavior where Lshift and Rshift are treated as valid different hotkeys. I can understand wanting to accept either when doing a combination. Can we allow both use-cases by default?

**Comment by YoshiRulz** [comment]
The workaround given in https://github.com/TASEmulators/BizHawk/issues/3184#issuecomment-1072055197 also allows just LShift or just RShift. I will properly integrate it into the UI before release.

**Comment by TheMetroid7998** [comment]
Wait, so you can't have LShift and RShift accepted at the same time as different bindings?
(Also is the next release going to fix the other issue I stated earlier?)

**Comment by YoshiRulz** [comment]
I'm not sure what issue you're talking about; I quickly tried binding a hotkey to `LeftShift,RightShift` and that worked. Please [grab a dev build](https://gitlab.com/TASVideos/BizHawk/-/jobs/2410326233/artifacts/download), uncheck the new setting:
![Screenshot_20220504_131924](https://user-images.githubusercontent.com/13409956/166619080-39412b2b-3835-4ba9-b471-9d84752e69a9.png)
...and report any remaining bugs.

**Comment by vadosnaprimer** [comment]
> I'm not sure what issue you're talking about; I quickly tried binding a hotkey to `LeftShift,RightShift` and that worked. Please [grab a dev build](https://gitlab.com/TASVideos/BizHawk/-/jobs/2410326233/artifacts/download), uncheck the new setting: ...and report any remaining bugs.

Reporting:

- Open a game and press LeftShift+F1 and then RightShift+F1. Both will work the same and save a state (as expected, because they are merged by default).
- Untick the checkbox in the Customize dialog, agree to change bindings automatically.
- Neither LeftShift+F1 nor RightShift+F1 will save a state anymore.
- Go to hotkey bindings and assign LeftShift to something (I assigned it to Pause).
- LeftShift+F1 will now save a state.

Same problem going the other way around. This is in latest dev and 2.10.
